src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerObserver.cs

```diff
@@ -38,9 +40,31 @@ public Task OpenAsync(ChangeFeedObserverContext context)
             return Task.CompletedTask;
         }
 
-        public Task ProcessChangesAsync(ChangeFeedObserverContext context, IReadOnlyList<Document> docs)
+        public async Task ProcessChangesAsync(ChangeFeedObserverContext context, IReadOnlyList<Document> docs)
         {
-            return this.executor.TryExecuteAsync(new TriggeredFunctionData() { TriggerValue = docs }, CancellationToken.None);
+            int retries = 0;
+
+            while (true)
+            {
+                var result = await this.executor.TryExecuteAsync(new TriggeredFunctionData() { TriggerValue = docs }, CancellationToken.None);
```

Comment:
It may be useful to pass the retry count here. That'd let the function decide if it wants to write out the document to some "poison" location for later processing if its on the last attempt.

Reply:
I'll take a look at this.